### PR TITLE
Convert all Experiment's components to standalone

### LINKED ISSUE #24759: [UI] - Convert all modules of Experiments to Standalone
### Parent Issue

_No response_

### User Story

As a developer, I want to standardize the components of the Experiment Feature to all Standalone components.

### Acceptance Criteria

- All the component needs to be Standalone

### Proposed Objective

Code Maintenance

### Proposed Priority

Priority 4 - Trivial

### External Links... Slack Conversations, Support Tickets, Figma Designs, etc.

_No response_

### Assumptions & Initiation Needs

_No response_

### Quality Assurance Notes & Workarounds

_No response_

### Sub-Tasks & Estimates

_No response_

### PR DESCRIPTION
**Pull request by oidacra**
### Proposed Changes
* Convert all Experiment's components to standalone

### Checklist
- [x] Tests
- [ ] Translations
- [ ] Security Implications Contemplated (add notes if applicable)

### Additional Info
** any additional useful context or info **

### Summary
🚚🛠️🚀

Refactored the dot-experiments components to be standalone and reusable in different contexts. Moved the components from the dot-experiments folder to the dot-edit-page folder. Updated the routing modules to use the loadComponent and loadChildren properties to dynamically load the components. Removed the unnecessary dot-experiments-shell module and routing module. Added the dot-experiments-shell-routes file to define the routes for the dot-experiments-shell component.

> _Sing, O Muse, of the mighty refactorings_
> _That the skilled developers wrought on the code_
> _To make the `dot-experiments` components_
> _Standalone and lazy, like Hermes the god._

### Walkthrough
*  Refactored dot-experiments components to be standalone and reusable in different contexts ([link](https://github.com/dotCMS/core/pull/24763/files?diff=unified&w=0#diff-63aad294dbf1fabe5880d6b2569bcb2a4425a1bbc9e084347586887e33e92ff5L3-R45), [link](https://github.com/dotCMS/core/pull/24763/files?diff=unified&w=0#diff-381d554f847a281a9ce615daebf585d34cf99c1126a5d2cb43a57e13331eb2b3R5), [link](https://github.com/dotCMS/core/pull/24763/files?diff=unified&w=0#diff-1302a6fcf9bb694e53e1544320e59b35f096b26db3626c1be72680d96ac69571L3-R8), [link](https://github.com/dotCMS/core/pull/24763/files?diff=unified&w=0#diff-67cf33540a0c27736adee88d2c206533e56a293b7f680574a806e457d9f5ced6L1-R9), [link](https://github.com/dotCMS/core/pull/24763/files?diff=unified&w=0#diff-67cf33540a0c27736adee88d2c206533e56a293b7f680574a806e457d9f5ced6L12-R39), [link](https://github.com/dotCMS/core/pull/24763/files?diff=unified&w=0#diff-4806b7552de8c71e943d59e12ff4c23033ccadbec690b6dc80f8f7064f3786afL2-R20), [link](https://github.com/dotCMS/core/pull/24763/files?diff=unified&w=0#diff-e3c81d7630ba385b59b6384acc495783a25f7f60689819c473f19b32dd35d36aL4-R10), [link](https://github.com/dotCMS/core/pull/24763/files?diff=unified&w=0#diff-e3c81d7630ba385b59b6384acc495783a25f7f60689819c473f19b32dd35d36aL16-R56), [link](https://github.com/dotCMS/core/pull/24763/files?diff=unified&w=0#diff-e3c81d7630ba385b59b6384acc495783a25f7f60689819c473f19b32dd35d36aL36), [link](https://github.com/dotCMS/core/pull/24763/files?diff=unified&w=0#diff-9c4b0835486cfc2775a1b34a7af5b629f3f99e8dfeb78873f5d7d669cf773106L2-R20), [link](https://github.com/dotCMS/core/pull/24763/files?diff=unified&w=0#diff-d1039b3e0c6659defbb304c91468c614e5ab6a5b34cdc7e501da1178d8422d77L1-R1), [link](https://github.com/dotCMS/core/pull/24763/files?diff=unified&w=0#diff-d1039b3e0c6659defbb304c91468c614e5ab6a5b34cdc7e501da1178d8422d77L17-R18))
*  Removed dot-experiments-list and dot-experiments-shell modules and routing modules, as they are no longer needed ([link](https://github.com/dotCMS/core/pull/24763/files?diff=unified&w=0#diff-20e2d9d83619338b684089d5f16a09a0098df411b03f8e807e425c2b9a1be32a), [link](https://github.com/dotCMS/core/pull/24763/files?diff=unified&w=0#diff-13e7957c2302896554e36d0e5be690c6d2fb3fe7e9b66ba17b14f639c2192b3c), [link](https://github.com/dotCMS/core/pull/24763/files?diff=unified&w=0#diff-7495cb0f51afb5db071b68f06683ebdd78f408491d326bd2b8eb295249c3309b), [link](https://github.com/dotCMS/core/pull/24763/files?diff=unified&w=0#diff-7893a5deac6017492855b801cf9e4e6ee352f594af36605382642136e02e88f4))
*  Added dot-experiments-shell-routes file to define the routes for the dot-experiments-shell component, which is the parent component for the dot-experiments feature ([link](https://github.com/dotCMS/core/pull/24763/files?diff=unified&w=0#diff-023b11c9275c9a6a8c1bbddbe3105f24e270e3bd7f48a29ea474e99a106775f0R1-R37))
*  Updated dot-edit-page route to use the dot-experiments-shell-routes instead of the dot-experiments-shell module, to enable lazy loading of the dot-experiments components ([link](https://github.com/dotCMS/core/pull/24763/files?diff=unified&w=0#diff-e085106ab179853e265adf11699afca1c0e2e8a51ea0a89fc3b1755bc25276cfL61-R63))
*  Removed dot-experiments-list component template, as the component uses the loadComponent property instead of the template property to dynamically load the appropriate child component based on the route parameters ([link](https://github.com/dotCMS/core/pull/24763/files?diff=unified&w=0#diff-42811b5baec7dd47a5151dd918f6bd3e6bdfde3312a1cdc09eadd91736b96691L7))
*  Updated dot-experiments-list-store class to use the route snapshot data to get the page title, as the route data is now populated by the dot-edit-page-resolver service ([link](https://github.com/dotCMS/core/pull/24763/files?diff=unified&w=0#diff-dd61b1a694006551d4ca399e82db92102beb8c7682d30c454f9ecc428478b040L340-R340))
*  Updated dot-experiments index file to export the dot-experiments-reports-routes instead of the dot-experiments-shell module, as the dot-experiments-reports feature is now lazy loaded and uses the loadChildren property instead of the loadComponent property ([link](https://github.com/dotCMS/core/pull/24763/files?diff=unified&w=0#diff-9383791961eef09685c93d4003cb40f261922530a6ad966fa61c6024eae53dd1L1-R1))

### CODE REVIEW

**Review by zJaaal — APPROVED** [review]

File: core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-configuration/dot-experiments-configuration.component.ts [hunk 1]
```diff
@@ -1,18 +1,48 @@
...
+        NgIf,
+        // dotCMS
+        DotMessagePipeModule,
+        DotExperimentsUiHeaderComponent,
+        DotExperimentsExperimentSummaryComponent,
+        DotExperimentsConfigurationVariantsComponent,
+        DotExperimentsConfigurationGoalsComponent,
+        DotExperimentsConfigurationTrafficComponent,
+        DotExperimentsConfigurationSchedulingComponent,
+        DotExperimentsConfigurationSkeletonComponent,
```
Comment (fmontes) [inline core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-configuration/dot-experiments-configuration.component.ts:40]: Is very clear these are dot :D

File: core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-list/dot-experiments-list.component.ts [hunk 2]
```diff
@@ -13,15 +17,41 @@ import {
...
 } from '@portlets/dot-experiments/dot-experiments-list/store/dot-experiments-list-store';
+import { DotExperimentsUiHeaderComponent } from '@portlets/dot-experiments/shared/ui/dot-experiments-header/dot-experiments-ui-header.component';
 import { DotDynamicDirective } from '@portlets/shared/directives/dot-dynamic.directive';
 
+const DOT_EXPERIMENT_LIST_COMPONENTS = [
+    DotExperimentsListSkeletonComponent,
+    DotExperimentsEmptyExperimentsComponent,
+    DotExperimentsStatusFilterComponent,
+    DotExperimentsListTableComponent
+];
```
Comment (fmontes) [inline core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-list/dot-experiments-list.component.ts:38]: Why?

File: core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-list/store/dot-experiments-list-store.ts [hunk 3]
```diff
@@ -337,7 +337,7 @@ export class DotExperimentsListStore
             ...initialState,
             page: {
                 pageId: route.snapshot.params.pageId,
-                pageTitle: route.snapshot.parent?.parent?.parent?.parent.data?.content?.page?.title
+                pageTitle: route.snapshot.parent?.parent?.parent?.data?.content?.page?.title
```
Comment (fmontes) [inline core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-list/store/dot-experiments-list-store.ts:340]: 😒

**Review by fmontes — DISMISSED** [review]

File: core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-list/dot-experiments-list.component.ts (hunk 2 above)
Comment (oidacra) [reply core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-list/dot-experiments-list.component.ts:38]: Testing, but I will use directly like the others. Thank.

File: core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-list/store/dot-experiments-list-store.ts (hunk 3 above)
Comment (oidacra) [reply core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-list/store/dot-experiments-list-store.ts:340]: Let me try to find a solution to this.

**Review by rjvelazco — APPROVED** [review]

**Review by hmoreras — APPROVED** [review]

**Pull request merged**